parquet-hadoop/src/main/java/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -202,12 +205,29 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
 
     while (!recordFound) {
       // no more records left
-      if (current >= total) { return false; }
+      if (current >= total) { return false;}
 
       try {
         checkRead();
-        currentValue = recordReader.read();
+
+        corruptRecordCounter.incRecordsSeen();
+        boolean corrupt = false;
+
+        try {
```

Comment:
Nested try catch block and multiple if...then blocks in this while loop make it hard to read and track. Also the inErrorsSeen depends on the calling of incRecordSeen is error prone. I think your propose is to have an enhanced recordReader that can track the corrupted record. So an Adapter Pattern like CorruptedTrackingRecordReader maybe able to simplify this logic a bit.